### DISCUSSION TITLE
Passing node from database to XSLT as a parameter

### DISCUSSION THREAD

**Discussion by kosek** [eXist-db/exist#4912] [Q&A]
Hi,

I'm trying passing nodes from DB to transformation as a parameter, e.g.:
```xquery
xquery version "3.1";

let $stylesheet := <xsl:stylesheet version="3.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
    <xsl:param name="p"/>
    <xsl:template name="start">
      <xsl:sequence select="$p"/>
    </xsl:template>
</xsl:stylesheet>

return
    transform(map { 
                'stylesheet-node': $stylesheet,
                'initial-template': QName('', 'start'),
                'stylesheet-params': map {
                  QName('', 'p'): /some/xpath/there
                }
              })?output

``` 
But I'm getting the following error message:

> class org.exist.dom.persistent.NodeProxy cannot be cast to class org.w3c.dom.Node (org.exist.dom.persistent.NodeProxy is in unnamed module of loader java.net.URLClassLoader @421faab1; org.w3c.dom.Node is in module java.xml of loader 'bootstrap')

How it is possible to access documents stored in exist-db inside transformation?

**Comment by kosek** [comment]
And more strangely same errors appear when I don't use parameters but I'm trying to load source of XSLT transformation from db using

````
'stylesheet-location': 'xmldb:exist:///db/apps/ibcompare/app/compare2html.xsl'
````

**Comment by dizzzz** [comment]
The entire transformation is done by saxon, which is an external library to eXist and there challenges pop-in; "xmldb:exist" is there not known despite our present custom URL protocol resolver.

**Comment by line-o** [comment]
@kosek there is an open PR which will likely fix this issue https://github.com/eXist-db/exist/pull/5882
